lightning/src/onion_message/messenger.rs

```diff
@@ -1047,18 +1081,37 @@ where
 	/// enqueueing any response for sending.
 	pub fn handle_onion_message_response<T: OnionMessageContents>(
 		&self, response: ResponseInstruction<T>
-	) {
-		if let ResponseInstruction::WithoutReplyPath(response) = response {
-			let message_type = response.message.msg_type();
-			let _ = self.find_path_and_enqueue_onion_message(
-				response.message, Destination::BlindedPath(response.reply_path), None,
-				format_args!(
-					"when responding with {} to an onion message with path_id {:02x?}",
-					message_type,
-					response.path_id
-				)
-			);
-		}
+	) -> Result<Option<SendSuccess>, SendError> {
+		let (response, create_reply_path) = match response {
+			ResponseInstruction::WithReplyPath(response) => (response, true),
+			ResponseInstruction::WithoutReplyPath(response) => (response, false),
+			ResponseInstruction::NoResponse => return Ok(None),
+		};
+
+		let message_type = response.message.msg_type();
+		let reply_path = if create_reply_path {
+			match self.create_blinded_path() {
+				Ok(reply_path) => Some(reply_path),
+				Err(err) => {
+					log_trace!(
+						self.logger,
+						"Failed to create reply_path when responding with {} to an onion message \
+						with path_id {:02x?}: {:?}",
+						message_type, response.path_id, err
+					);
+					return Err(err);
```

Comment:
Let's add a test for the error case.

Reply:
Introduce a new test in [27899b1](https://github.com/lightningdevkit/rust-lightning/commit/27899b1a04bfd78db34f2c22ad489f1a6fd4e5e4) that conditionally checks for the failure case!
Thanks for the pointer!